### PR TITLE
[14.0][ADD] account_payment_batch_process

### PR DESCRIPTION
**Pull request by Hardik-OSI**
Todo

- [x] Invoice Reconciliation
- [x] pre-commit

### CODE REVIEW

**Comment by max3903** [comment]
@Hardik-OSI Please run:
```
pip install pre-commit
pre-commit run -a
```
and fix the errors on this module. Try to split the 2 complex methods, that should help you reduce the complexity below 16.
https://github.com/OCA/account-payment/pull/391/checks?check_run_id=1936718128#step:4:121

**Comment by SerenaCyr1** [comment]
After adding payments to the batch process, the pay amount should populate the balance amount.
payamt

**Comment by SerenaCyr1** [comment]
Also, after creating the batch manually and clicking Make Payments, you receive an error.

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/odoo/src/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/odoo/src/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/odoo/src/odoo/http.py", line 359, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/odoo/src/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/odoo/src/odoo/http.py", line 347, in checked_call
    result = self.endpoint(*a, **kw)
  File "/odoo/src/odoo/http.py", line 912, in __call__
    return self.method(*args, **kw)
  File "/odoo/src/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
  File "/odoo/src/addons/web/controllers/main.py", line 1381, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/odoo/external-src/third-party/printnode_base/controllers/main.py", line 51, in _call_kw
    return super(DataSetProxy, self)._call_kw(model, method, args, kwargs)
  File "/odoo/src/addons/web/controllers/main.py", line 1369, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/odoo/src/odoo/api.py", line 396, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/odoo/src/odoo/api.py", line 383, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/odoo/external-src/account-payment/account_payment_batch_process/wizard/account_payment_register.py", line 622, in make_payments
    + str(data_get.invoice_id.number)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/odoo/src/odoo/http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/odoo/src/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
AttributeError: 'account.move' object has no attribute 'number'
```

**Comment by Hardik-OSI** [comment]
> After adding payments to the batch process, the pay amount should populate the balance amount.
> payamt

Can you give me the step to reproduce this issue?

**Comment by max3903** [comment]
@Hardik-OSI Enter a value in "Batch Payment Total" and then click on "Auto-fill Pay Amount" button.

CC: @SerenaCyr1

**Comment by bodedra** [comment]
Please remove  account_payment_batch_process/changes.txt

File: account_payment_batch_process/views/account_move.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    <record model="ir.ui.menu" id="account_batch_payment.menu_batch_payment_purchases">
+        <field name="action" eval="False" />
+    </record>
+
+    <record model="ir.ui.menu" id="account_batch_payment.menu_batch_payment_sales">
+        <field name="action" eval="False" />
+    </record>
+
+    <!-- <delete model="ir.actions.act_window" id="account.action_account_payment_from_invoices"/> -->
```
Comment (bodedra) [inline account_payment_batch_process/views/account_move.xml]: Commented code can be remove.

**Comment by max3903** [comment]
@Hardik-OSI The reconciliation is done and we get the expected functional behavior.

Please fix the pre-commit errors and make sure all the checks are green. Schedule a call with me if you get stuck.

CC: @hiren-serpentcs

File: account_payment_batch_process/views/account_move.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <!--  Supplier move tree -->
+    <record id="batch_move_supplier_tree" model="ir.ui.view">
+        <field name="name">account.move.batch.supplier.tree</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_in_invoice_tree" />
+        <field name="arch" type="xml">
+            <field name="state" position="replace" />
+            <field name="partner_id" position="after">
+                <field name="state" />
+            </field>
```
Comment (i-vyshnevska) [inline account_payment_batch_process/views/account_move.xml]:
```suggestion
            <field name="partner_id" position="after">
                <field name="state"  position="move"/>
            </field>
```

File: account_payment_batch_process/wizard/account_payment_register.py [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+
+    def get_payment_values(self, group_data=None):
+        res = {}
+        if group_data:
+            writeoff_amount = 0
+            writeoff_account_id = False
+            writeoff_name = False
+            for invoice_id in list(group_data["inv_val"]):
+                values = group_data["inv_val"][invoice_id]
+                if not self.currency_id.is_zero(values["payment_difference"]) \
```
Comment (i-vyshnevska) [inline account_payment_batch_process/wizard/account_payment_register.py]:
```suggestion
                if self.currency_id and not self.currency_id.is_zero(values["payment_difference"]) \
```

**Comment by dreispt** [comment]
@Hardik-OSI @max3903 Is there an expectation to complete this work?

**Comment by smangukiya** [comment]
@murtuzasaleh Can you please work on this and complete this? You are already forecast for next week on this so. Thank you

CC: @dreispt @max3903

**Review by max3903 — APPROVED** [review]

**Comment by enterlineconnor** [comment]
Upon trying to click the "FILE SUCCESSFULLY UPLOADED" button with multiple different res_partners on the pay

Screen Shot 2021-06-22 at 9 33 59 AM

any help would be appreciated!

**Review by dreispt — APPROVED** [review]

**Pull request merged**